Remove direct setup.py usage

bwoodsend opened the pull request:
Following on from https://github.com/pyinstaller/pyinstaller/pull/8779#issuecomment-2336796001:

* `python setup.py sdist` is now `python -m build --sdist`
* `python setup.py bdist_wheel` is now `pip wheel .` or `python -m build --wheel`
* Cross building is now done by setting the `PYI_WHEEL_TAG` and `PYI_PLATFORM` environment variables before building a wheel
* `python setup.py bdist_wheels` is now a new script `./release/build-wheels`

bwoodsend committed:
setup: Fix +x permission on release script

bwoodsend committed:
Stop appending git commit hashes to PyInstaller.__version__

Setuptools long ago switched to a static, bytecode scanning methond for
extracting the value of __version__ from __init__.py [1]. This means
that our __version__ += git hash never makes it into the package's
metadata.

The only time when the tag adding mechanism has any effect at all is
when PyInstaller is ran from an editable installation of a git clone
from which anyone can already get the hash using git log/show.

.. [1]: https://github.com/pypa/setuptools/blob/v74.1.2/setuptools/depends.py#L101-L164

bwoodsend committed:
Remove unused (and long broken) test Docker container

bwoodsend committed:
ci: Remove unused release workflow

bwoodsend committed:
Remove direct setup.py usage

* `python setup.py sdist` is now `python -m build --sdist`
* `python setup.py bdist_wheel` is now `pip wheel .` or
  `python -m build --wheel`
* Cross building is now done by setting the `PYI_WHEEL_TAG` and
  `PYI_PLATFORM` environment variables before building a wheel
* `python setup.py bdist_wheels` is now a new script
  `./release/build-wheels`

bwoodsend committed:
release: Remove multiarch/qemu-user-static from bootloader build script

Between distributions rolling their own ways to configure binfmt,
rootless podman and docker desktop, using the official docker image to
set the kernel's map of executable headers to qemu emulators seems to
break more setups than it fixes.

bwoodsend commented:
Contrary to what I said in https://github.com/pyinstaller/pyinstaller/pull/8779#issuecomment-2336822991, we can use `python -m build` but only if we specify `--wheel`. With `--wheel`, it builds directly from source code. Without it, it builds an sdist (which excludes bootloaders) then builds a wheel from the sdist, forcing it to recompile the bootloaders (for the native platform rather than the one we want).

I wouldn't bet on this loophole existing forever though. They might change it to build from an sdist then throw the sdist away at then end if it's not wanted.

bwoodsend commented:
Mind if we swap back over to flake8? Ruff's crazy file size and fast release cadence combined with my network rate makes the switch a bit of an anti-optimisation?

rokm commented:
> Mind if we swap back over to flake8? Ruff's crazy file size and fast release cadence combined with my network rate makes the switch a bit of an anti-optimisation?

Sure, I have no real preference regarding `flake8` vs `ruff`.

rokm commented on PyInstaller/__init__.py:
```diff
@@ -26,24 +25,6 @@
-        # and 'setup.py' was not called with 'sdist' argument. For creating source tarball we do not want git revision
-        # in the filename.
-        try:
-            __version__ += get_repo_revision()
```
I think this was the only use of `PyInstaller.utils._gitrevision` (which supplies git commit/revision information for archive snapshots generated via git-archive), so we can remove that as well. (Plus the corresponding [entry in .gitattributes](https://github.com/pyinstaller/pyinstaller/blob/5bd9dd70b4c792103d048ecf43933ba8d71e0993/.gitattributes#L25))

rokm commented on PyInstaller/utils/git.py:
```diff
@@ -1,59 +0,0 @@
-    cwd = os.path.dirname(gitdir)
-    if not path.exists(gitdir):
-        try:
-            from PyInstaller.utils._gitrevision import rev
```
I think this was the only use of `PyInstaller.utils._gitrevision` (which supplies git commit/revision information for archive snapshots generated via git-archive), so we can remove that as well. (Plus the corresponding [entry in .gitattributes](https://github.com/pyinstaller/pyinstaller/blob/5bd9dd70b4c792103d048ecf43933ba8d71e0993/.gitattributes#L25))

bwoodsend committed:
Restore flake8, abandon ruff

The only meaningful difference I feel ruff makes is how much more time I
spend waiting for it to download (both due to its wheel size and its
release cadence). The runtime performance boosts are unnoticeable.

This reverts commit 3dadde82a4162d3b0ec59e713eebbae1430a14c7 (#7559).

bwoodsend committed:
Remove flake8 from test requirements (tests/requirements-tools.txt)

This matches how yapf is handled. Having it in the
requirements-tools.txt means that it gets redundantly installed into the
Docker container used for testing Alpine.

rokm commented on tests/unit/test_winmanifest.py:
```diff
@@ -548,7 +548,7 @@ def test_manifest_write_to_exe_non_ascii_characters(tmp_path):
   </trustInfo>
 </assembly>
-""".encode("utf-8") # noqa: E501
+""".encode("utf-8") # noqa: E128,E261,E501
```
I think E261 here is required only because the noqa comment itself is missing two whitespaces?

The other four manifest strings do not seem to require E261 at all.

rokm commented on tests/unit/test_winmanifest.py:
```diff
@@ -225,7 +225,7 @@ def test_manifest_custom_manifest_no_trust_info():
   </dependency>
 </assembly>
-"""  # noqa: E501
+"""  # noqa: E128,E261,E501
```
E261 is redundant

bwoodsend replied:
Ugh, one of those days...

rokm commented on tests/unit/test_winmanifest.py:
```diff
@@ -302,7 +302,7 @@ def test_manifest_custom_manifest_no_ms_common_controls():
   </application>
 </assembly>
-"""  # noqa: E501
+"""  # noqa: E128,E261,E501
```
E261 is redundant

rokm commented on tests/unit/test_winmanifest.py:
```diff
@@ -377,7 +377,7 @@ def test_manifest_custom_manifest_different_ms_common_controls():
   </dependency>
 </assembly>
-"""  # noqa: E501
+"""  # noqa: E128,E261,E501
```
E261 is redundant

rokm commented on tests/unit/test_winmanifest.py:
```diff
@@ -452,7 +452,7 @@ def test_manifest_custom_manifest_no_ms_common_controls_with_custom_dependency()
   </dependency>
 </assembly>
-"""  # noqa: E501
+"""  # noqa: E128,E261,E501
```
E261 is redundant
